hsmd/hsmd_wire.csv

```diff
@@ -356,6 +356,15 @@ msgdata,hsmd_sign_message,msg,u8,len
 msgtype,hsmd_sign_message_reply,123
 msgdata,hsmd_sign_message_reply,sig,secp256k1_ecdsa_recoverable_signature,
 
+# sign a raw message with a derived key
+msgtype,hsmd_sign_message_with_key,45
```

Comment:
Can we call this bip137_sign_message?  We're going to want to do 332 eventually, though it's a complex mess and I'm not sure I even understand it :(

Reply:
Agree, that's a better naming.
Also I am not very happy with the RPC name "signmessagewithkey", we don't provide a key in the arguments,
but it couldn't be "signmessagewithaddress" because addresses are not able to sign *per se*.

Reply:
[Fixed](https://github.com/ElementsProject/lightning/compare/cbc9205bdc39a0285fd055fa3082466c31941b54..0fcdb2ac5796bd62bd0d0b17f4f7b8696a27a9aa)